[inductor] Clean up AOTInductor runtime interface

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109673
* #109606
* #109450
* #109498
* __->__ #109436

Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

desertfire committed:
[inductor] Clean up AOTInductor runtime interface

Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

chenyang78 reviewed (approved):
LGTM. Thanks!

ezyang commented on torch/csrc/inductor/aot_runtime/interface.h:
```diff
@@ -64,58 +53,61 @@ AOTInductorError AOTInductorModelContainerDelete(
     AOTInductorModelContainerHandle container_handle,
     size_t input_idx,
-    AOTInductorParamShape* input_shape);
+    const int64_t** ret_input_sizes,
```
This is a preexisting condition, but what's the ownership on the returned `int*`?

desertfire replied:
It is borrow, so are other `ret_*`. I can add comments to state that.

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Clean up AOTInductor runtime interface"


Summary: Break the use of the AOTInductorParamShape struct in more primitive types; Change the naming of the returned parameters to be consistent with the aoti_torch ones. The actual at::Tensor clean up will come next.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire commented:
Landed with an exported combined PR, https://github.com/pytorch/pytorch/pull/109678
